packages/valory/skills/termination_abci/rounds.py

```diff
@@ -57,7 +59,7 @@ class SynchronizedData(BaseSynchronizedData):
     """
 
     @property
-    def safe_contract_address(self) -> Optional[str]:
+    def safe_contract_address(self) -> Optional[str]:  # type: ignore
```

Comment:
why not removed? it's available on base now

Reply:
As I mentioned in f5d052115626e341099caabe478f8c71669403d7, It is still overridden here, because it is necessary for the `termination_abci` to function correctly. In essence, we cannot assume that the registration round is over when the `safe_contract_address` is called by the `TerminationRound` since it runs concurrently, from the beginning of the app.

@0xArdi can provide more insight about this.

Reply:
As @Adamantios said, it might happen that this is accessed before registration is over.

Reply:
This doesn’t explain why we need it. Now the address is a pre condition so the property as defined on the base class is sufficient. It can never be None

Reply:
Yes, this sounds reasonable since the data will now contain the safe contract (must be in the setup data) from the db's initialization!

https://github.com/valory-xyz/open-autonomy/blob/10b215f6848c7bc25ca178038f4678cc57fca0db/packages/valory/skills/abstract_round_abci/base.py#L610-L612

Reply:
Done in 3d6d752e68f2d51c02150b023be951e6ec2d9e6c.